### ISSUE TITLE
Bug: Frontend Build and Deploy Github Action is broken

### ISSUE THREAD

**Issue by jbubar** [hackforla/VRMS#1603]
### Dependency:
On hold until Vite is implemented.

### Overview

The Github Action that we use to deploy the frontend to dev (and most likely prod) is broken. It gets caught on the Build & Push Image to ECR step maybe having something to do with running an old version of node. This is really important because we cannot currently push changes to prod or dev until we fix this.

![Screen Shot 2024-03-02 at 12 29 20 AM](https://github.com/hackforla/VRMS/assets/53061723/3615bfe1-deb0-40f7-b40d-7961ce913f51)

### Action Items
- [ ] Do some debugging
    - [ ] Figure out if it is our node version
    - [ ] Could it be recent changes to the repo?
    - [ ] Could it be this merge in the incubator repo? https://github.com/hackforla/incubator/pull/28
- [ ] Propose a solution in the vrms team meeting 

### Resources/Instructions
[Frontend Github Action](https://github.com/hackforla/VRMS/actions/workflows/aws-frontend-deploy.yml)
<details>
    <summary>
Stack Trace
</summary>

![Screen Shot 2024-03-02 at 12 26 06 AM](https://github.com/hackforla/VRMS/assets/53061723/99ebf09d-baf6-47c4-8d0c-321daa25e2b9)

``` bash



##[debug]Evaluating condition for step: 'Build & Push Image to ECR'
##[debug]Evaluating: success()
##[debug]Evaluating success:
##[debug]=> true
##[debug]Result: true
##[debug]Starting: Build & Push Image to ECR
##[debug]Loading inputs
##[debug]Evaluating: secrets.INCUBATOR_AWS_ACCESS_KEY_ID
##[debug]Evaluating Index:
##[debug]..Evaluating secrets:
##[debug]..=> Object
##[debug]..Evaluating String:
##[debug]..=> 'INCUBATOR_AWS_ACCESS_KEY_ID'
##[debug]=> '***'
##[debug]Result: '***'
##[debug]Evaluating: secrets.INCUBATOR_AWS_SECRET_ACCESS_KEY
##[debug]Evaluating Index:
##[debug]..Evaluating secrets:
##[debug]..=> Object
##[debug]..Evaluating String:
##[debug]..=> 'INCUBATOR_AWS_SECRET_ACCESS_KEY'
##[debug]=> '***'
##[debug]Result: '***'
##[debug]Evaluating: secrets.INCUBATOR_AWS_ACCOUNT_ID
##[debug]Evaluating Index:
##[debug]..Evaluating secrets:
##[debug]..=> Object
##[debug]..Evaluating String:
##[debug]..=> 'INCUBATOR_AWS_ACCOUNT_ID'
##[debug]=> '***'
##[debug]Result: '***'
##[debug]Evaluating: needs.setup_env.outputs.AWS_APPENV
##[debug]Evaluating Index:
##[debug]..Evaluating Index:
##[debug]....Evaluating Index:
##[debug]......Evaluating needs:
##[debug]......=> Object
##[debug]......Evaluating String:
##[debug]......=> 'setup_env'
##[debug]....=> Object
##[debug]....Evaluating String:
##[debug]....=> 'outputs'
##[debug]..=> Object
##[debug]..Evaluating String:
##[debug]..=> 'AWS_APPENV'
##[debug]=> 'vrms-client-prod'
##[debug]Result: 'vrms-client-prod'
##[debug]Evaluating: env.AWS_REGION
##[debug]Evaluating Index:
##[debug]..Evaluating env:
/usr/bin/docker run --name d[2](https://github.com/hackforla/VRMS/actions/runs/8108999100/job/22183779171#step:7:2)c120ddc66b691d42ff85e0fdda7ffd0[3](https://github.com/hackforla/VRMS/actions/runs/8108999100/job/22183779171#step:7:3)[4](https://github.com/hackforla/VRMS/actions/runs/8108999100/job/22183779171#step:7:4)0_ee4fb1 --label 72d2c1 --workdir /github/workspace --rm -e "AWS_SHARED_CLUSTER" -e "AWS_APP_NAME" -e "AWS_REGION" -e "DOCKERFILE" -e "DOCKER_PATH" -e "AWS_DEFAULT_REGION" -e "AWS_ACCESS_KEY_ID" -e "AWS_SECRET_ACCESS_KEY" -e "INPUT_ACCESS_KEY_ID" -e "INPUT_SECRET_ACCESS_KEY" -e "INPUT_ACCOUNT_ID" -e "INPUT_REPO" -e "INPUT_REGION" -e "INPUT_TAGS" -e "INPUT_DOCKERFILE" -e "INPUT_PATH" -e "INPUT_ASSUME_ROLE" -e "INPUT_CREATE_REPO" -e "INPUT_SET_REPO_POLICY" -e "INPUT_REPO_POLICY_FILE" -e "INPUT_EXTRA_BUILD_ARGS" -e "INPUT_PREBUILD_SCRIPT" -e "HOME" -e "GITHUB_JOB" -e "GITHUB_REF" -e "GITHUB_SHA" -e "GITHUB_REPOSITORY" -e "GITHUB_REPOSITORY_OWNER" -e "GITHUB_REPOSITORY_OWNER_ID" -e "GITHUB_RUN_ID" -e "GITHUB_RUN_NUMBER" -e "GITHUB_RETENTION_DAYS" -e "GITHUB_RUN_ATTEMPT" -e "GITHUB_REPOSITORY_ID" -e "GITHUB_ACTOR_ID" -e "GITHUB_ACTOR" -e "GITHUB_TRIGGERING_ACTOR" -e "GITHUB_WORKFLOW" -e "GITHUB_HEAD_REF" -e "GITHUB_BASE_REF" -e "GITHUB_EVENT_NAME" -e "GITHUB_SERVER_URL" -e "GITHUB_API_URL" -e "GITHUB_GRAPHQL_URL" -e "GITHUB_REF_NAME" -e "GITHUB_REF_PROTECTED" -e "GITHUB_REF_TYPE" -e "GITHUB_WORKFLOW_REF" -e "GITHUB_WORKFLOW_SHA" -e "GITHUB_WORKSPACE" -e "GITHUB_ACTION" -e "GITHUB_EVENT_PATH" -e "GITHUB_ACTION_REPOSITORY" -e "GITHUB_ACTION_REF" -e "GITHUB_PATH" -e "GITHUB_ENV" -e "GITHUB_STEP_SUMMARY" -e "GITHUB_STATE" -e "GITHUB_OUTPUT" -e "RUNNER_DEBUG" -e "RUNNER_OS" -e "RUNNER_ARCH" -e "RUNNER_NAME" -e "RUNNER_ENVIRONMENT" -e "RUNNER_TOOL_CACHE" -e "RUNNER_TEMP" -e "RUNNER_WORKSPACE" -e "ACTIONS_RUNTIME_URL" -e "ACTIONS_RUNTIME_TOKEN" -e "ACTIONS_CACHE_URL" -e "ACTIONS_RESULTS_URL" -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/_temp/_runner_file_commands":"/github/file_commands" -v "/home/runner/work/VRMS/VRMS":"/github/workspace" 72d2c1:20ddc66b691d42ff8[5](https://github.com/hackforla/VRMS/actions/runs/8108999100/job/22183779171#step:7:5)e0fdda[7](https://github.com/hackforla/VRMS/actions/runs/8108999100/job/22183779171#step:7:7)ffd0340
== START LOGIN
Traceback (most recent call last):
  File "/usr/bin/aws", line 1[9](https://github.com/hackforla/VRMS/actions/runs/8108999100/job/22183779171#step:7:9), in <module>
    import awscli.clidriver
  File "/usr/lib/python2.7/site-packages/awscli/clidriver.py", line 17, in <module>
    import botocore.session
  File "/usr/lib/python2.7/site-packages/botocore/session.py", line 29, in <module>
    import botocore.configloader
  File "/usr/lib/python2.7/site-packages/botocore/configloader.py", line 19, in <module>
    from botocore.compat import six
  File "/usr/lib/python2.7/site-packages/botocore/compat.py", line 27, in <module>
    from dateutil.tz import tzlocal
  File "/usr/lib/python2.7/site-packages/dateutil/__init__.py", line 5, in <module>
    from ._version import version as __version__
  File "/usr/lib/python2.7/site-packages/dateutil/_version.py", line [10](https://github.com/hackforla/VRMS/actions/runs/8108999100/job/22183779171#step:7:10)
    version: str
           ^
SyntaxError: invalid syntax
##[debug]Docker Action run completed with exit code 1
##[debug]Finishing: Build & Push Image to ECR
```

</details>

**Comment by trillium** [comment]
<img width="1102" alt="image" src="https://github.com/hackforla/VRMS/assets/5898009/86f8a576-950b-40b0-88be-85a7e17b6b86">

Not sure why it is now succeeding, but it built without changing anything

**Comment by JackHaeg** [comment]
**Next Steps:** 
- [ ] Meeting with DevOps team
- [ ] Doing more debugging
  - [ ] (@jbubar to add steps)

**Comment by jbubar** [comment]
Looks like it magically works again!! hmmm

**Comment by jbubar** [comment]
We are placing it in the icebox.. someone should come in and see if they can figure out why it broke and what changed

**Comment by JackHaeg** [comment]
- Closing issue as new frontend / backend rebuild issues have arisen since the creation of this issue. New issue available here: #1767
